SCloudRandomizer/SearchParamsVC.m

```diff
@@ -53,22 +58,37 @@ - (void)didReceiveMemoryWarning {
 - (IBAction)done:(id)sender {
     [self dismissKeyboard];
     
-    if (![Utility stringIsNilOrEmpty:self.txtKeywords.text])
+    if(!([self.txtBpmTo.text isEqualToString:@""]) && ([self.txtBpmTo.text intValue] < [self.txtBpmFrom.text intValue]))
     {
-        NSNumberFormatter *formatter = [[NSNumberFormatter alloc] init];
-        self.searchParams.hasChanged = YES;
-        self.searchParams.keywords = self.txtKeywords.text;
-        self.searchParams.lowBpm = [formatter numberFromString:self.txtBpmFrom.text];
-        self.searchParams.highBpm = [formatter numberFromString:self.txtBpmTo.text];
-        
-        [self dismissViewControllerAnimated:YES completion:nil];
+        UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Bad BPM Range" message:@"Please make sure the BPM range is valid!" delegate:self cancelButtonTitle:@"Ok" otherButtonTitles:nil];
+        [alert show];
+        self.txtBpmTo.text = @"";
+        [self.txtBpmTo becomeFirstResponder];
+    }
+    else if(!([self.txtDurationTo.text isEqualToString:@""]) && ([self.txtDurationTo.text intValue] < [self.txtDurationFrom.text intValue]))
+    {
```

Comment:
you could have one helper method that has this alert logic that takes in a title, message and whatever object needs to be made first responder?